Add support for Advanced Search

fretje opened the pull request:
This one needs https://github.com/fullstackhero/dotnet-webapi-boilerplate/pull/384

@iammukeshm Is it something like this you had in mind with regards to advanced search on individual columns?

I have only implemented it on Products now... if it's ok, I would integrate it into the EntityTable...

iammukeshm commented:
Yeah, looks good. So this would generate the checkboxes dynamically and send those params via the advancedSearch property right?

fretje commented:
Yep, I've already moved it to the entitytable. It takes only the properties that are defined in the fields array (and uses the sortlabel property to send the field to the server).

I've also added advanced filtering for product... Not sure if that's the best way (using property setters to refresh the server content). Enlighten me with how that should be done properly! ;-)
